src/test/java/testsuite/integration/AwsIamAuthenticationIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+    private static final String TEST_DB_USER = System.getenv("TEST_DB_USER");
+    private static final String DB_CONN_STR = DB_CONN_STR_PREFIX + TEST_DB_CLUSTER_IDENTIFIER + DB_READONLY_CONN_STR_SUFFIX;
+
+    /**
+     * Attempt to connect using the wrong database username
+     */
+    @Test
+    public void test_1_WrongDatabaseUsername() {
+        final Properties props = initProp("WRONG_" + TEST_DB_USER + "_USER", TEST_PASSWORD);
+
+        Assertions.assertThrows(
+            SQLException.class,
+            () -> {
+                final Connection conn = DriverManager.getConnection(DB_CONN_STR, props);
+            }
+        );
+    }
+
+    /**
+     * Attempt to connect without specifying a database username
+     */
+    @Test
+    public void test_2_NoDatabaseUsername() {
+        final Properties props = initProp("", TEST_PASSWORD);
+
+        Assertions.assertThrows(
+            SQLException.class,
+            () -> {
+                final Connection conn = DriverManager.getConnection(DB_CONN_STR, props);
+            }
+        );
+    }
+
+    /**
+     * Attempt to connect using IP address instead of a hostname.
+     */
+    @Test
+    public void test_3_UsingIPAddress() {
+        final Properties props = initProp(TEST_DB_USER, TEST_PASSWORD);
+
+        final String hostname = DB_CONN_STR.substring(DB_CONN_STR_PREFIX.length());
+        Assertions.assertThrows(
+            SQLException.class,
+            () -> { final Connection conn = DriverManager.getConnection(DB_CONN_STR_PREFIX + hostToIP(hostname), props);
+        });
+    }
+
+    /**
+     * Attempt to connect using valid database username/password & valid Amazon RDS hostname.
+     */
+    @Test
+    public void test_4_ValidConnectionProperties() throws SQLException {
+        final Properties props = initProp(TEST_DB_USER, TEST_PASSWORD);
+
+        final Connection conn = DriverManager.getConnection(DB_CONN_STR, props);
+        Assertions.assertNotNull(conn);
+
+        final Statement myQuery = conn.createStatement();
+        final ResultSet rs = myQuery.executeQuery("SELECT 1;");
+        while (rs.next()) {
+            Assertions.assertEquals("1", rs.getString(1));
+        }
+
+        conn.close();
+        Assertions.assertTrue(conn.isClosed());
+    }
+
+    /**
+     * Attempt to connect using valid database username, valid Amazon RDS hostname, but no password.
+     */
+    @Test
+    public void test_5_ValidConnectionPropertiesNoPassword() throws SQLException {
+        final Properties props = initProp(TEST_DB_USER, "");
+        final Connection conn = DriverManager.getConnection(DB_CONN_STR, props);
+        Assertions.assertNotNull(conn);
+
+        final Statement myQuery = conn.createStatement();
+        final ResultSet rs = myQuery.executeQuery("SELECT 1;");
+        while (rs.next()) {
+            Assertions.assertEquals("1", rs.getString(1));
+        }
+
+        conn.close();
+        Assertions.assertTrue(conn.isClosed());
```

Comment:
I think asserting `connection is not null` is enough.

Reply:
I think it is enough because the goal of this test is: ensuring we can create a connection with valid properties.